### PR TITLE
Command: also print removed env vars

### PR DESCRIPTION
**Pull request by RalfJung**
There is no real shell syntax for unsetting an env var so easily, so we have to make one up. But we already do that for showing the 'program' name so I hope that's okay here, too. No strong opinion on what that should look like, I went with `unset(VAR_NAME)` for now.

### CODE REVIEW

**Comment by the8472** [comment]
https://unix.stackexchange.com/a/388849 suggests wrapping a command in a subshell. It's gnarly syntax but it would be copy-pastable.

> (unset ENV_VAR; exec command)

**Comment by joshtriplett** [comment]
I do feel like being able to easily copy-paste the command is a desirable feature. Not worth substantially sacrificing readability for, though.

**Comment by the8472** [comment]
The subshell syntax could be applied conditionally only when an environment variable was unset. For adding variables the prefix approach is sufficient.

**Comment by RalfJung** [comment]
We already have `cd` which alters the surrounding environment, so I don't think we need a subshell just for `unset`.

Or conversely, if we want to worry about subshells, we should also do it for `cd`.

**Comment by joshtriplett** [comment]
@RalfJung Good call. Having it as an executable command seems like the important part; the subshell doesn't seem worth it, and as you said we already alter the surrounding environment. If someone pastes in the command, they can *see* that it has `unset` and `cd` lines in it, and if they don't want that they can wrap it themselves easily enough.

**Comment by joshtriplett** [comment]
Since this has a noticeable change in user-visible behavior, let's doublecheck consensus:

@rfcbot merge

**Comment by the8472** [comment]
I don't think this really needs an FCP since the behavior was changed #97176 not too long ago without much fanfare, but ok

**Comment by RalfJung** [comment]
@the8472 do you know if there is an easy way to express "clear the entire environment" in unix commands?

I'm thinking maybe I want to represent that as well. It could be something like `clearenv && FOO=bar cmd args`? That doesn't actually work but it should be clear enough, hopefully?

**Comment by the8472** [comment]
since clearing the environment means that no variable needs to be explicitly unset anymore (since none are set) it can be expressed this way:

```
$ env -i FOO=BAR BAZ=QUUX cat /proc/self/environ
FOO=BARBAZ=QUUX
```

**Comment by RalfJung** [comment]
Ah, that is neat.

**Comment by RalfJung** [comment]
In fact we could use `env -u FOO BAR="val" prog` to represent unsetting in a way that doesn't alter the current shell...

File: library/std/src/sys/unix/process/process_common.rs [hunk 1]
```diff
@@ -558,6 +558,24 @@ impl fmt::Debug for Command {
             if let Some(ref cwd) = self.cwd {
                 write!(f, "cd {cwd:?} && ")?;
             }
+            if self.env.does_clear() {
+                write!(f, "env -i ")?;
+                // Altered env vars will be printed next, that should exactly work as expected.
+            } else {
+                // Removed env vars need a separate command.
+                // We use a single `unset` command for all of them.
```
Comment (coolreader18) [inline library/std/src/sys/unix/process/process_common.rs:566]: This comment seems out of date.

Comment (RalfJung) [reply library/std/src/sys/unix/process/process_common.rs]: Good catch, thanks!

**Pull request merged**